### PR TITLE
Add recipe for Poolsnp

### PR DESCRIPTION
**Pull request by abhilesh**
Describe your pull request here

PoolSNP is a heuristic SNP caller, which uses an MPILEUP file and a reference genome in FASTA format as inputs ([Github repo](https://github.com/capoony/PoolSNP)). 
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/poolsnp/build.sh
Comment (martin-g) [inline recipes/poolsnp/build.sh:14]:
```suggestion
mkdir -p ${PREFIX}/bin
```

File: recipes/poolsnp/build.sh
Comment (martin-g) [inline recipes/poolsnp/build.sh:18]: This is not really needed. You did it already at line 13 (https://github.com/bioconda/bioconda-recipes/pull/49156/files#diff-3180f716065e95d04e19dba02fcb40838a15acf2f2d2772cf255515c2ed19a48R13)

File: recipes/poolsnp/meta.yaml
Comment (martin-g) [inline recipes/poolsnp/meta.yaml:8]: Could you please ask the authors for a release/tag ?

File: recipes/poolsnp/meta.yaml
Comment (martin-g) [inline recipes/poolsnp/meta.yaml:13]:
```suggestion
    - python
```

File: recipes/poolsnp/meta.yaml
Comment (martin-g) [inline recipes/poolsnp/meta.yaml:16]:
```suggestion
    - python
```
Py2k is no more supported anyway. At the moment Bioconda builds with 3.8. 3.9, 3.10, 3.11 and 3.12.

File: recipes/poolsnp/meta.yaml
Comment (martin-g) [inline recipes/poolsnp/meta.yaml:1]:
```suggestion
{% set version = "0.1.0" %}
```
I'd suggest to start with 0.x since there is no official release of https://github.com/capoony/PoolSNP yet

File: recipes/poolsnp/meta.yaml
Comment (martin-g) [inline recipes/poolsnp/meta.yaml:43]: 👍🏻

**Comment by aliciaaevans** [comment]
Sorry for the useless bot comments. It's supposed to only look for artifacts and comment if the checks all succeed.

File: recipes/poolsnp/build.sh
Comment (abhilesh) [reply recipes/poolsnp/build.sh:29]: Thanks for the suggestion, I followed the style to create all necessary directories

File: recipes/poolsnp/build.sh
Comment (abhilesh) [reply recipes/poolsnp/build.sh]: Removed duplicate entries

File: recipes/poolsnp/meta.yaml
Comment (abhilesh) [reply recipes/poolsnp/meta.yaml]: [Asked the authors for a release](https://github.com/capoony/PoolSNP/issues/10) - Awaiting response, else I will maintain a fork of the repo with releases

File: recipes/poolsnp/meta.yaml
Comment (abhilesh) [reply recipes/poolsnp/meta.yaml]: Made the necessary modifications

File: recipes/poolsnp/meta.yaml
Comment (abhilesh) [reply recipes/poolsnp/meta.yaml]: Updated version numbers

**Comment by abhilesh** [comment]
@BiocondaBot please add label

File: recipes/poolsnp/meta.yaml
Comment (abhilesh) [reply recipes/poolsnp/meta.yaml]: Updated references to python

**Comment by abhilesh** [comment]
@martin-g Thanks a lot for your comments, incorporating the tests was giving me a lot of trouble, finally managed to solve it by adding tests separately in the `run-test.sh` file. 

Would you like me to squash the commits for a cleaner history?

**Comment by martin-g** [comment]
> Would you like me to squash the commits for a cleaner history?

No need. We use `Squash and commit` Github functionality.
The only problem here is the usage of Git SHA instead of a release/tag.
I'll let others decide whether this is a stopper or not.

**Comment by abhilesh** [comment]
@martin-g, got a release/tag from the author and updated the recipe to use the release/tag instead of the Git SHA.

File: recipes/poolsnp/build.sh
Comment (martin-g) [inline recipes/poolsnp/build.sh:5]: `BASE_DIR` is the same as the already provided `RECIPE_DIR`

**Review by martin-g — APPROVED** [review]

File: recipes/poolsnp/build.sh
Comment (abhilesh) [reply recipes/poolsnp/build.sh:5]: I needed the path for the work directory where the github files are downloaded to set the relative paths for the `scripts/` and the `TestData/` directories. 

`RECIPE_DIR` returns the root of the recipes directory, whereas `BASE_DIR` here returns the root of the work directory for `conda-bld`

```
'BASE_DIR is set to: $HOME/miniconda3/conda-bld/poolsnp_1721646158691/work'
'RECIPE_DIR is set to: $HOME/bioconda-recipes/recipes/poolsnp'
```

**Comment by abhilesh** [comment]
Thank you for all your help in reviewing this PR! Before I delete the branch, I want to ensure that the tool would be available as a biocontainer as well. From the [biocontainers docs](https://biocontainers-edu.readthedocs.io/en/master/what_is_biocontainers.html#:~:text=After%20the%20PR%20gets%20merged%2C%20a%20Conda%20package%20gets%20created%20and%20the%20corresponding%20docker%20container%20get%20pushed%20into%20Quay.io%20Registry%20and%20the%20BioContainers%20Registry), I understand that the containers are automatically built after the PR for the conda recipe is merged. I can find the PoolSNP package on [bioconda repo](https://anaconda.org/bioconda/poolsnp), but can't find it in the biocontainers repo. Are there any additional steps to be taken to make it available on biocontainers?

I also see that some checks were unsuccessful for the [merge commit](https://github.com/bioconda/bioconda-recipes/commit/39da4e0fc4f3d9c6b483b5aed40c058586b415c2). Specifically, the `build_and_test-osx-arm64` test for CircleCI shows up as cancelled. However, clicking on the test takes me to the details for the [`rebuid-salmon` branch ](https://app.circleci.com/pipelines/github/bioconda/bioconda-recipes/123276/workflows/63862eb2-2089-4973-8cdc-57c61c8b52cf/jobs/198417/steps). Not sure, if that's correct?

**Pull request merged**
